### ISSUE TITLE
NuGet osx-arm64 package broken

### ISSUE THREAD

**Issue by pederlh** [google/or-tools#3881]
**What version of OR-Tools and what language are you using?**
Version: 9.5.2237.0
Language: C#

**Which solver are you using (e.g. CP-SAT, Routing Solver, GLOP, BOP, Gurobi)**
N/A

**What operating system (Linux, Windows, ...) and version?**
MacOS Ventura 13.5

**What did you do?**
I have been trying to use OR-Tools in a C# .NET project, but when running there is always an error with loading a file/assembly. The error shows up even when trying to run the [example project](https://developers.google.com/optimization/install/dotnet/pkg_mac) linked to in the [install guide](https://developers.google.com/optimization/install/dotnet/pkg_mac). 

The example project gives the following error message:

```bash
pederlh@Peders-MacBook-Air:~/Repositories/dotnet_or-tools$ dotnet run -c release
/usr/local/share/dotnet/sdk/7.0.302/Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/pederlh/.nuget/packages/google.ortools/9.5.2237/lib/net6.0/Google.OrTools.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/pederlh/Repositories/dotnet_or-tools/project.csproj]
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Google.OrTools, Version=9.5.2237.0, Culture=neutral, PublicKeyToken=53dbb1ea090cbef7'. The system cannot find the file specified.

File name: 'Google.OrTools, Version=9.5.2237.0, Culture=neutral, PublicKeyToken=53dbb1ea090cbef7'
```

In addition, following the [guide on building from source](https://developers.google.com/optimization/install/dotnet/source_mac) also leads to failure, and gives a similar error message

```
pederlh@Peders-MacBook-Air:~/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534$ make test
/Library/Developer/CommandLineTools/usr/bin/make run SOURCE=examples/3_jugs_regular/3_jugs_regular.cs
"/opt/homebrew/bin/dotnet" build examples/3_jugs_regular/project.csproj
MSBuild version 17.6.1+8ffc3fe3d for .NET
  Determining projects to restore...
/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj : warning NU1701: Package 'Google.OrTools 9.6.2534' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project.
  Restored /Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj (in 1,72 sec).
/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj : warning NU1701: Package 'Google.OrTools 9.6.2534' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [TargetFramework=netcoreapp3.1]
/usr/local/share/dotnet/sdk/7.0.302/Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/pederlh/.nuget/packages/google.ortools/9.6.2534/lib/net46/Google.OrTools.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj::TargetFramework=netcoreapp3.1]
/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj : warning NU1701: Package 'Google.OrTools 9.6.2534' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [TargetFramework=net6.0]
/usr/local/share/dotnet/sdk/7.0.302/Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/pederlh/.nuget/packages/google.ortools/9.6.2534/lib/net6.0/Google.OrTools.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj::TargetFramework=net6.0]
  project -> /Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/bin/Debug/netcoreapp3.1/Google.OrTools.3_jugs_regular.dll
  project -> /Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/bin/Debug/net6.0/Google.OrTools.3_jugs_regular.dll

Build succeeded.

/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj : warning NU1701: Package 'Google.OrTools 9.6.2534' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project.
/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj : warning NU1701: Package 'Google.OrTools 9.6.2534' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [TargetFramework=netcoreapp3.1]
/usr/local/share/dotnet/sdk/7.0.302/Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/pederlh/.nuget/packages/google.ortools/9.6.2534/lib/net46/Google.OrTools.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj::TargetFramework=netcoreapp3.1]
/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj : warning NU1701: Package 'Google.OrTools 9.6.2534' was restored using '.NETFramework,Version=v4.6.1, .NETFramework,Version=v4.6.2, .NETFramework,Version=v4.7, .NETFramework,Version=v4.7.1, .NETFramework,Version=v4.7.2, .NETFramework,Version=v4.8, .NETFramework,Version=v4.8.1' instead of the project target framework '.NETCoreApp,Version=v3.1'. This package may not be fully compatible with your project. [TargetFramework=net6.0]
/usr/local/share/dotnet/sdk/7.0.302/Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/pederlh/.nuget/packages/google.ortools/9.6.2534/lib/net6.0/Google.OrTools.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/project.csproj::TargetFramework=net6.0]
    5 Warning(s)
    0 Error(s)

Time Elapsed 00:00:03.13
"/opt/homebrew/bin/dotnet" run --no-build --framework net6.0 --project examples/3_jugs_regular/project.csproj --
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'. The system cannot find the file specified.

File name: 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'
   at ThreeJugsRegular.Solve(Int32 n)
   at ThreeJugsRegular.Main(String[] args) in /Users/pederlh/Repositories/or-tools_arm64_macOS-13.2.1_dotnet_v9.6.2534/examples/3_jugs_regular/3_jugs_regular.cs:line 278
make[1]: *** [run] Error 134
```

**Comment by Mizux** [comment]
Few issue here:

1. Why your platform is `MSIL` ? (should be `osx-arm64` or `osx-x64` imho)
ref: https://learn.microsoft.com/en-us/dotnet/api/microsoft.build.utilities.processorarchitecture.msil

2. Can you give me the output of `dotnet --info`
you should have something like this
```sh
%dotnet --info
.NET SDK:
 Version:   7.0.304
 Commit:    7e794e2806
Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.5
 OS Platform: Darwin
 RID:         osx.13-arm64 // NOTICE THE RID HERE
 Base Path:   /usr/local/share/dotnet/sdk/7.0.304/
Host:
  Version:      7.0.7
  Architecture: arm64 // AND ARCHITECTURE HERE
  Commit:       5b20af47d9
.NET SDKs installed:
  7.0.304 [/usr/local/share/dotnet/sdk]
.NET runtimes installed:
  Microsoft.AspNetCore.App 7.0.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 7.0.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
Other architectures found:
  None
Environment variables:
  Not set
global.json file:
  Not found
Learn more:
  https://aka.ms/dotnet/info
Download .NET:
  https://aka.ms/dotnet/download
```

3. My bad https://github.com/or-tools/dotnet_or-tools is still in v9.5, you can try to change
https://github.com/or-tools/dotnet_or-tools/blob/69a9f11b12cd9f9b6a0579458d5e3d8958870a6c/project.csproj#L45 to set `9.6.*` instead

**Comment by pederlh** [comment]
Thanks for the quick reply!

1. I'm not 100% sure about the `MSIL` platform, but I think it has to do with the `.csproj` file using the default Any CPU target platform. If I add
```xml
<PropertyGroup>
   <PlatformTarget>x64</PlatformTarget>
</PropertyGroup>
```
to the project the warning disappears, but the code still doesn't run.
```bash
pederlh@Peders-MacBook-Air:~/Repositories/dotnet_or-tools$ dotnet run -c Release
Unhandled exception. System.IO.FileLoadException: Could not load file or assembly 'Google.OrTools.BasicExample, Version=1.0.0.0, Culture=neutral, PublicKeyToken=null'.
```
https://learn.microsoft.com/en-us/visualstudio/msbuild/errors/msb3270?view=vs-2022
https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-options/output#platformtarget

2.
```bash
pederlh@Peders-MacBook-Air:~/Repositories/dotnet_or-tools$ dotnet --info
.NET SDK:
 Version:   7.0.302
 Commit:    990cf98a27

Runtime Environment:
 OS Name:     Mac OS X
 OS Version:  13.5
 OS Platform: Darwin
 RID:         osx.13-arm64
 Base Path:   /usr/local/share/dotnet/sdk/7.0.302/

Host:
  Version:      7.0.5
  Architecture: arm64
  Commit:       8042d61b17

.NET SDKs installed:
  6.0.408 [/usr/local/share/dotnet/sdk]
  7.0.202 [/usr/local/share/dotnet/sdk]
  7.0.302 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 6.0.16 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.4 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.5 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 6.0.16 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.4 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.5 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

Other architectures found:
  x64   [/usr/local/share/dotnet/x64]
    registered at [/etc/dotnet/install_location_x64]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
```

3. Changing the version number to `9.6.*` still doesn't help. When not editing the target platform, I get
```bash
pederlh@Peders-MacBook-Air:~/Repositories/dotnet_or-tools$ dotnet run -c Release
/usr/local/share/dotnet/sdk/7.0.302/Microsoft.Common.CurrentVersion.targets(2352,5): warning MSB3270: There was a mismatch between the processor architecture of the project being built "MSIL" and the processor architecture of the reference "/Users/pederlh/.nuget/packages/google.ortools/9.6.2534/lib/net6.0/Google.OrTools.dll", "AMD64". This mismatch may cause runtime failures. Please consider changing the targeted processor architecture of your project through the Configuration Manager so as to align the processor architectures between your project and references, or take a dependency on references with a processor architecture that matches the targeted processor architecture of your project. [/Users/pederlh/Repositories/dotnet_or-tools/project.csproj]
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'. The system cannot find the file specified.

File name: 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'
```

**Comment by Mizux** [comment]
some test locally:

1. need to use `dotnet nuget locals all --clear` between test to clear dotnet package cache
note: also use `rm -rf bin obj` to clean up project

2. Seems to work with `9.4.*`
```sh
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %dotnet build -c Release
[0]─[~/work/repo/dotnet_or-tools]-[main]
[*v*]─me@host %otool -L  /Users/me/.nuget/packages/google.ortools.runtime.osx-arm64/9.4.1874/runtimes/osx-arm64/native/libortools.9.dylib
/Users/me/.nuget/packages/google.ortools.runtime.osx-arm64/9.4.1874/runtimes/osx-arm64/native/libortools.9.dylib:
	@rpath/libortools.9.dylib (compatibility version 9.0.0, current version 9.4.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1858.112.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.23.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1311.100.3)
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %dotnet run -c Release     
Number of variables = 2
Number of constraints = 1
Solution:
Objective value = 4
x = 1
y = 1
```

but fail with 9.5 or 9.6....
```sh
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %dotnet build -c Release
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %otool -L  /Users/corentinl/.nuget/packages/google.ortools.runtime.osx-arm64/9.*/runtimes/osx-arm64/native/libortools.9.dylib
/Users/corentinl/.nuget/packages/google.ortools.runtime.osx-arm64/9.6.2534/runtimes/osx-arm64/native/libortools.9.dylib:
	@rpath/libortools.9.dylib (compatibility version 9.0.0, current version 9.6.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1953.255.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1300.36.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.0.0)
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %dotnet run -c Release
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'. The system cannot find the file specified.

File name: 'Google.OrTools, Version=9.6.2534.0, Culture=neutral, PublicKeyToken=37e4c4e778bf1216'
```

still investigating...

**Comment by Mizux** [comment]
Last test using or-tools main branch to build an osx-arm64 "only" package then hacking project RessourcePath to find them

project.csproj:
```xml
 <!-- Dependencies -->
  <PropertyGroup>
    <RestoreSources>../../main/export;$(RestoreSources);https://api.nuget.org/v3/index.json</RestoreSources>
  </PropertyGroup>

  <ItemGroup>
    <Compile Include="BasicExample.cs" />
    <PackageReference Include="Google.OrTools" Version="9.7.*" />
  </ItemGroup>
```


```sh
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %otool -L  /Users/corentinl/.nuget/packages/google.ortools.runtime.osx-arm64/9.*/runtimes/osx-arm64/native/libortools.9.dylib
/Users/corentinl/.nuget/packages/google.ortools.runtime.osx-arm64/9.7.2996/runtimes/osx-arm64/native/libortools.9.dylib:
	@rpath/libortools.9.dylib (compatibility version 9.0.0, current version 9.7.0)
	/System/Library/Frameworks/CoreFoundation.framework/Versions/A/CoreFoundation (compatibility version 150.0.0, current version 1971.0.0)
	/usr/lib/libc++.1.dylib (compatibility version 1.0.0, current version 1500.65.0)
	/usr/lib/libSystem.B.dylib (compatibility version 1.0.0, current version 1319.100.3)
[0]─[~/work/repo/dotnet_or-tools]-[main]
[^v^]─me@host %dotnet run -c Release                                                                                                       
Number of variables = 2
Number of constraints = 1
Solution:
Objective value = 4
x = 1
y = 1
```

**Comment by Mizux** [comment]
seems related to:
* https://github.com/google/or-tools/issues/3661

**Comment by pederlh** [comment]
Using `9.4.*` seems like a quick temp fix 👍🏼

**Comment by Mizux** [comment]
Investigating v9.7

If replacing the google.ortools.dll from the meta package from nuget by the one built locally only for macos arm64, it will works...

pseudo code

```sh
# clean all package cache
$ dotnet nuget locals all --clear

$ cd dotnet_ortools # hack project.csproj to use `9.7.*`
$ dotnet run -c Release # this will pull the nuget package


cd ~/.nuget/packages/google.ortools/9.7.2996/lib/net6.0 

```

**Comment by mslazynski** [comment]
According to my experience, building a meta package on macos-aarm64 does not help. Still had to swap the `dll`.

**Comment by parveen-chhillar** [comment]
Unhandled exception. System.IO.FileNotFoundException: Could not load file or assembly 'Google.OrTools, Version=9.7.2996.0, Culture=neutral, PublicKeyToken=53dbb1ea090cbef7'. The system cannot find the file specified.


The same issue is on the latest version of Google Or Tools,  I'm using Mac Venture (13.5) with M2 Chip and Visual Studio 2022 with targeting framework 7.

While on the server root directory of the project, the Google.OrTools.dll file exists, but it still says File not found exception.

Any solution ?

**Comment by parveen-chhillar** [comment]
I'm using macOS 13.5 (Venture) with an M2 Chip, and I'm working with Visual Studio 2022 and .NET 7. I also have Google OR-Tools version 9.7.2996 installed. I encountered an issue, but I managed to resolve it. Here are the steps I followed:

1. Change your build settings from x64/x86 to Any CPU.

2. On the server, within your app service, navigate to the "Configuration" settings. Under "General Settings," locate the "Platform Setting" and switch it from 32-bit to 64-bit.

3. If the issue still persists, you can manually check if a DLL file exists on your local system at this path: `/Users/user/.nuget/packages/google.ortools/9.7.2996/lib/net6.0/Google.OrTools.dll`

4. If the DLL file doesn't exist at that location, you can try deleting the "packages" folder at this path: `/Users/user/.nuget/`

5. After deleting the "packages" folder, close your project and then reopen it. The "packages" folder should be recreated automatically, and you should find the DLL file in the specified path.

By following these steps, I was able to resolve my issue.

**Comment by Mizux** [comment]
Need to postpone to v9.9 (or at least didn't find why and how to fix it in a day)
My dev note:

v9.4
```sh
%unzip -l google.ortools.9.4.1874.nupkg 
Archive:  google.ortools.9.4.1874.nupkg
  Length      Date    Time    Name
---------  ---------- -----   ----
      504  2022-08-11 14:35   _rels/.rels
     2248  2022-08-11 14:35   Google.OrTools.nuspec
  1601536  2022-08-11 12:35   lib/net6.0/Google.OrTools.dll
   727743  2022-08-11 12:35   lib/net6.0/Google.OrTools.xml
  1601536  2022-08-11 12:35   lib/netcoreapp3.1/Google.OrTools.dll
   727743  2022-08-11 12:35   lib/netcoreapp3.1/Google.OrTools.xml
    13822  2022-06-23 14:40   content/bop/bop_parameters.proto
     1763  2022-08-09 09:51   content/constraint_solver/assignment.proto
      988  2022-08-09 09:51   content/constraint_solver/demon_profiler.proto
     6656  2022-06-23 14:40   content/constraint_solver/routing_enums.proto
    27591  2022-06-23 14:40   content/constraint_solver/routing_parameters.proto
     1444  2022-06-23 14:40   content/constraint_solver/search_limit.proto
     3275  2022-06-23 14:40   content/constraint_solver/search_stats.proto
     3851  2022-06-23 14:40   content/constraint_solver/solver_parameters.proto
    24218  2022-06-23 14:40   content/glop/parameters.proto
     3161  2022-06-23 14:40   content/graph/flow_problem.proto
    30232  2022-06-23 14:40   content/linear_solver/linear_solver.proto
    18209  2022-06-23 14:40   content/pdlp/solvers.proto
    17346  2022-06-23 14:40   content/pdlp/solve_log.proto
     4725  2022-06-23 14:40   content/sat/boolean_problem.proto
    34080  2022-06-23 14:40   content/sat/cp_model.proto
     1363  2022-06-23 14:40   content/sat/cp_model_service.proto
    63760  2022-08-09 09:51   content/sat/sat_parameters.proto
     1347  2022-06-23 14:40   content/util/optional_boolean.proto
     6141  2020-04-27 07:55   content/orLogo.png
      647  2022-08-11 14:35   [Content_Types].xml
      731  2022-08-11 14:35   package/services/metadata/core-properties/2bf5c396282845a48b853e4c4ae536e0.psmdcp
     9476  2022-08-16 09:23   .signature.p7s
---------                     -------
  4936136                     28 files
```
vs v9.7
```sh
unzip -l Google.OrTools.9.7.2996.nupkg 
Archive:  Google.OrTools.9.7.2996.nupkg
  Length      Date    Time    Name
---------  ---------- -----   ----
      508  2023-08-09 09:07   _rels/.rels
     2336  2023-08-09 09:07   Google.OrTools.nuspec
  1716736  2023-08-09 16:07   lib/net462/Google.OrTools.dll
   806496  2023-08-09 16:07   lib/net462/Google.OrTools.xml
  1716736  2023-08-09 16:07   lib/net6.0/Google.OrTools.dll
   806496  2023-08-09 16:07   lib/net6.0/Google.OrTools.xml
    14109  2023-05-31 15:37   content/bop/bop_parameters.proto
     1832  2023-05-31 15:37   content/constraint_solver/assignment.proto
     1017  2023-02-20 12:20   content/constraint_solver/demon_profiler.proto
     6798  2023-02-20 12:20   content/constraint_solver/routing_enums.proto
    30381  2023-07-28 09:34   content/constraint_solver/routing_parameters.proto
     1483  2023-02-20 12:20   content/constraint_solver/search_limit.proto
     3519  2023-03-01 07:50   content/constraint_solver/search_stats.proto
     3991  2023-02-20 12:20   content/constraint_solver/solver_parameters.proto
    24913  2023-03-01 07:50   content/glop/parameters.proto
     3241  2023-03-01 07:50   content/graph/flow_problem.proto
    31046  2023-05-31 15:37   content/linear_solver/linear_solver.proto
    25874  2023-07-28 09:34   content/pdlp/solvers.proto
    21088  2023-05-31 15:37   content/pdlp/solve_log.proto
     4838  2023-02-20 12:20   content/sat/boolean_problem.proto
    35329  2023-05-31 15:37   content/sat/cp_model.proto
     1401  2023-02-20 12:20   content/sat/cp_model_service.proto
    73588  2023-07-28 09:34   content/sat/sat_parameters.proto
     1381  2023-02-20 12:20   content/util/optional_boolean.proto
     6141  2023-02-20 12:20   content/orLogo.png
      656  2023-08-09 09:07   [Content_Types].xml
      761  2023-08-09 09:07   package/services/metadata/core-properties/d353d211770446bbbf0dd6c3872c018c.psmdcp
---------                     -------
  5342695                     27 files
```

**Comment by Mizux** [comment]
May have found the bug
https://github.com/google/or-tools/blob/a33bf9a8d80711f4bb80199aa88d64506cfb8812/cmake/dotnet.cmake#L49-L53

When building a Multi-platform package (i.e. `-DUNIVERSAL_DOTNET_PACKAGE=ON`) we should **not** set the platform
(or set it to `AnyCPU`)
https://github.com/google/or-tools/blob/a33bf9a8d80711f4bb80199aa88d64506cfb8812/cmake/dotnet.cmake#L346-L356
https://github.com/google/or-tools/blob/a33bf9a8d80711f4bb80199aa88d64506cfb8812/cmake/dotnet.cmake#L365-L368

On my way to test it...

**Comment by Mizux** [comment]
1) v9.7 meta package has been built on windows VM to have net462 support for windows (unity) user.

2) On macos M1 (arm64) using the pet project https://www.or-tools.dev/dotnet_or-tools (using v9.7)
`dotnet run` or `dotnet run -r osx-arm64`: will fail
note: playing with `~/.nuget/package` and ortools v9.7 v9.8LM I manage to get the error:
```

```

3) Using  `dotnet run -r osx-x64` worked even if only a arm64 dotnet-sdk is installed !
note: I thing under the hood rosetta2 compat manage to execute the osx-x64 byte code.

4) I've created a v9.8LM having only linux-x64 and osx-arm64 runtime and the meta has been built on Linux with `/p:Platform=x64`
My guess is on unix `\p:Platform` is simply ignored so when running on osx-arm64 it still work...
note: using `dotnet run -r osx-x64` with v9.8LM I got a error wihch is expected since this version only provide osx-arm64 rid.
```

```

5) On my way to create a v9.8WM (only windows-x64 and osx-arm64 rid + meta build on windows) and play with the /p:Platform flag....
